src/preact.d.ts

```diff
@@ -1,349 +1,414 @@
+// Type definitions for Preact v7.0
+// TypeScript Version: 2.2
+
+export = preact;
+export as namespace preact;
+
 declare namespace preact {
-	interface ComponentProps<C extends Component<any, any> | FunctionalComponent<any>> {
-		children?:JSX.Element[];
-		key?:string | number | any;
-		ref?:(el: C) => void;
+
+	type PreactType = string | ComponentClass<any, any> | SFC<any>;
+	type Key = string | number;
+	type Ref<T> = ((instance: T) => any);
+	type ComponentState = {} | void;
+	interface Attributes {
+		key?: Key;
+	}
+	interface ClassAttributes<T> extends Attributes {
+		ref?: Ref<T>;
+	}
+
+	/**
+	 * @deprecated.
+	 * This is no longer necessary to use directly as we have SFC type and Component
+	 *
+	 * This was used to allow clients to pass `ref` and `key`
+	 * to `createElement`, which is no longer necessary due to intersection
+	 * types. If you need to declare a props object before passing it to
+	 * `createElement` or a factory, use `ClassAttributes<T>`:
+	 *
+	 * ```ts
+	 * var b: Button;
+	 * var props: ButtonProps & ClassAttributes<Button> = {
+	 *     ref: b => button = b, // ok!
+	 *     label: "I'm a Button"
+	 * };
+	 * ```
+	 */
+	interface ComponentProps {
+		children?: PreactChildren;
+		key?: Key;
+		ref?: Ref<any>;
 	}
 
 	interface DangerouslySetInnerHTML {
-	 __html: string;
+		__html: string;
 	}
 
 	interface PreactHTMLAttributes {
-		dangerouslySetInnerHTML?:DangerouslySetInnerHTML;
-		key?:string;
-		ref?:(el?: Element) => void;
+		dangerouslySetInnerHTML?: DangerouslySetInnerHTML;
+		key?: Key;
+		ref?: Ref<Element>;
```

Comment:
correct, string refs are only patched into preact-compat, preact core is just function refs.